### PR TITLE
Update to the new v12 api, and remove Table support

### PR DESCRIPTION
**Pull request by NiklasArbin**
Update to new dependencies

### CODE REVIEW

File: knightbus-azurestorage/src/KnightBus.Azure.Storage/StorageQueueClient.cs [hunk 1]
```diff
@@ -60,22 +60,21 @@ public static string GetDeadLetterName(string queueName)
...
                 .ContinueWith(task => _queue.DeleteMessageAsync(message.QueueMessageId, message.PopReceipt), TaskContinuationOptions.OnlyOnRanToCompletion)
                 .ConfigureAwait(false);
         }
 
         public async Task<int> GetDeadLetterCountAsync()
         {
-            await _dlQueue.FetchAttributesAsync().ConfigureAwait(false);
-            return _dlQueue.ApproximateMessageCount ?? 0;
+            var props = await _dlQueue.GetPropertiesAsync().ConfigureAwait(false);
+            return props.Value.ApproximateMessagesCount;
```
Comment (andrevirdarson) [inline knightbus-azurestorage/src/KnightBus.Azure.Storage/StorageQueueClient.cs:77]: Are we sure that the Value prop is always set?

File: knightbus-azurestorage/src/KnightBus.Azure.Storage/StorageQueueClient.cs [hunk 2]
```diff
@@ -117,12 +114,14 @@ public async Task AbandonByErrorAsync(StorageQueueMessage message, TimeSpan? vis
...
 
-            var cloudMessage = new CloudQueueMessage(_serializer.Serialize(storageMessage.Properties));
             try
             {
-                var blob = _container.GetBlockBlobReference(storageMessage.BlobMessageId);
-                await blob.UploadTextAsync(_serializer.Serialize(message)).ConfigureAwait(false);
-                await _queue.AddMessageAsync(cloudMessage, TimeSpan.MaxValue, delay ?? TimeSpan.Zero, null, null).ConfigureAwait(false);
+                var blob = _container.GetBlobClient(storageMessage.BlobMessageId);
+                using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(_serializer.Serialize(message))))
+                {
```
[the remark this answers is not in the document]
Comment (NiklasArbin) [reply knightbus-azurestorage/src/KnightBus.Azure.Storage/StorageQueueClient.cs:121]: Det ska inte behövas eftersom vi inte läst strömmen. Hade vi däremot rört strömmen efter upload behöver vi sätta 0

File: knightbus-azurestorage/src/KnightBus.Azure.Storage/StorageQueueClient.cs (hunk 1 above)
Comment (NiklasArbin) [reply knightbus-azurestorage/src/KnightBus.Azure.Storage/StorageQueueClient.cs:77]: From what I understand all operations are wrapped in the Azure.Response<> class, but if this fails then a RequestFailedException should be thrown

File: knightbus-azurestorage/src/KnightBus.Azure.Storage/Singleton/BlobLockManager.cs [hunk 3]
```diff
@@ -117,57 +110,58 @@ private static async Task WriteLeaseBlobMetadata(CloudBlockBlob blob, string lea
...
+                throw;
             }
 
             try
             {
-                await blob.UploadTextAsync(string.Empty).ConfigureAwait(false);
+                var bytes = Encoding.UTF8.GetBytes(string.Empty);
+                using (var stream = new MemoryStream(bytes))
+                {
+                    stream.Position = 0;
```
Comment (viktor-h) [inline knightbus-azurestorage/src/KnightBus.Azure.Storage/Singleton/BlobLockManager.cs:157]: I guess this one can be removed?

File: knightbus-azurestorage/src/KnightBus.Azure.Storage/Singleton/BlobLockManager.cs [hunk 4]
```diff
@@ -117,57 +110,58 @@ private static async Task WriteLeaseBlobMetadata(CloudBlockBlob blob, string lea
...
-        private static async Task<bool> TryCreateAsync(CloudBlockBlob blob, CancellationToken cancellationToken)
+        private static async Task<bool> TryCreateAsync(BlobClient blob, CancellationToken cancellationToken)
         {
             try
             {
-                await blob.UploadTextAsync(string.Empty).ConfigureAwait(false);
+                var bytes = Encoding.UTF8.GetBytes(string.Empty);
+                using (var stream = new MemoryStream(bytes))
+                {
+                    stream.Position = 0;
```
Comment (viktor-h) [inline knightbus-azurestorage/src/KnightBus.Azure.Storage/Singleton/BlobLockManager.cs:120]: I guess this one can be removed?

**Review by viktor-h — APPROVED** [review]

**Pull request merged**